astropy/modeling/fitting.py

```diff
@@ -415,19 +415,23 @@ def __call__(self, model, x, y, z=None, weights=None, **kwargs):
 
         fitted_model = self.fitter(model, x, y, z, weights, **kwargs)
         if z is None:
+            fitted_model = self.fitter(fitted_model, x, y, **kwargs)
```

Comment:
Yes you are right, but for some reason, I did not see that when updating the code.   I just saw what happened after the if statement.